apps/omg_child_chain/test/omg_child_chain/block_queue/core_test.exs

```diff
@@ -237,130 +237,186 @@ defmodule OMG.ChildChain.BlockQueue.CoreTest do
+    test "Block generation is driven by last enqueued block Ethereum height and whether block is empty", %{
       empty: empty
     } do
       assert {:dont_form_block, _} =
                empty
-               |> Core.set_ethereum_status(0, 0, false)
+               |> Core.set_ethereum_status(0, 0, 1)
 
       assert {:dont_form_block, _} =
                empty
-               |> Core.set_ethereum_status(1, 0, true)
+               |> Core.set_ethereum_status(1, 0, 0)
 
       assert {:do_form_block, queue} =
                empty
-               |> Core.set_ethereum_status(1, 0, false)
+               |> Core.set_ethereum_status(1, 0, 1)
 
       assert {:dont_form_block, _} =
                queue
                |> Core.enqueue_block("1", 1000, 0)
-               |> Core.set_ethereum_status(0, 0, false)
+               |> Core.set_ethereum_status(0, 0, 1)
 
       assert {:dont_form_block, _} =
                queue
                |> Core.enqueue_block("1", 1000, 1)
-               |> Core.set_ethereum_status(1, 0, false)
+               |> Core.set_ethereum_status(1, 0, 3)
 
       assert {:dont_form_block, _} =
                queue
                |> Core.enqueue_block("1", 1000, 0)
-               |> Core.set_ethereum_status(1, 0, true)
+               |> Core.set_ethereum_status(1, 0, 0)
 
       # Ethereum advanced since enqueue and block isn't empty -> order forming of next block
       assert {:do_form_block, queue} =
                queue
                |> Core.enqueue_block("1", 1000, 0)
-               |> Core.set_ethereum_status(1, 0, false)
+               |> Core.set_ethereum_status(1, 0, 2)
 
       # no enqueue
       assert {:dont_form_block, queue} =
                queue
-               |> Core.set_ethereum_status(1, 0, false)
+               |> Core.set_ethereum_status(1, 0, 1)
 
       assert {:dont_form_block, _} =
                queue
                |> Core.enqueue_block("2", 2000, 1)
-               |> Core.set_ethereum_status(1, 0, false)
+               |> Core.set_ethereum_status(1, 0, 1)
 
       assert {:do_form_block, _} =
                queue
                |> Core.enqueue_block("2", 2000, 1)
-               |> Core.set_ethereum_status(2, 0, false)
+               |> Core.set_ethereum_status(2, 0, 5)
     end
 
     test "if Ethereum progressed to no later where last enqueue happened, don't ask to form", %{empty: empty} do
       assert {:dont_form_block, queue} =
                empty
-               |> Core.set_ethereum_status(0, 0, false)
+               |> Core.set_ethereum_status(0, 0, 1)
                |> elem(1)
                |> Core.enqueue_block("1", 1000, 0)
                |> Core.enqueue_block("2", 2000, 1)
-               |> Core.set_ethereum_status(1, 2000, false)
+               |> Core.set_ethereum_status(1, 2000, 1)
+    end
+  end
+
+  describe "check_block_fulfilled/2" do
+    test "does not form block by default", %{empty: empty} do
```

Comment:
what's the purpose of this test? Shouldn't we reject txs if we get to `max_block_size`?

Reply:
By default decision to form a block is to align with Ethereum blocks (`block_submit_every_nth`) and block size isn't considered. I don't change default behavior, so here we test old expected behavior. 

And yes - new transaction above max block size are rejected - but its different flow.